query-algebrizer/src/clauses/convert.rs

```diff
@@ -51,6 +51,53 @@ macro_rules! coerce_to_typed_value {
     } }
 }
 
+pub trait ValueTypes {
+    fn potential_types(&self, schema: &Schema) -> Result<ValueTypeSet>;
+}
+
+impl ValueTypes for FnArg {
+    fn potential_types(&self, schema: &Schema) -> Result<ValueTypeSet> {
+        Ok(match self {
+                &FnArg::EntidOrInteger(x) => {
+                    if ValueType::Ref.accommodates_integer(x) {
+                        // TODO: also see if it's a valid entid?
+                        ValueTypeSet::of_longs()
+                    } else {
+                        ValueTypeSet::of_one(ValueType::Long)
+                    }
+                },
+
+                &FnArg::IdentOrKeyword(ref x) => {
+                    if schema.get_entid(x).is_some() {
```

Comment:
This bit in particular doesn't, except in the manner that _all_ of our queries could fundamentally change behavior if the schema changes!